src/gui/dialogs/DialogRadioButton.cpp

```diff
@@ -138,28 +201,139 @@ void RadioButton::draw_n_btns(const size_t btn_count) {
             buffer[length] = 0;
             drawn = string_view_utf8::MakeRAM((const uint8_t *)buffer);
         }
-        button_draw(splits[i], GetBackColor(), drawn, pfont, GetBtnIndex() == i && IsEnabled());
+        button_draw(splits[i], GetBackColor(), GetParent() ? GetParent()->GetBackColor() : GetBackColor(), drawn, pfont, GetBtnIndex() == i && IsEnabled());
     }
     color_t spaces_clr = (GetBackColor() == COLOR_ORANGE) ? COLOR_BLACK : COLOR_ORANGE;
     for (size_t i = 0; i < btn_count - 1; ++i) {
         display::FillRect(spaces[i], spaces_clr);
     }
 }
 
-void RadioButton::button_draw(Rect16 rc_btn, color_t back_color, string_view_utf8 text, const font_t *pf, bool is_selected) {
+void RadioButton::button_draw(Rect16 rc_btn, color_t back_color, color_t parent_color, string_view_utf8 text, const font_t *pf, bool is_selected) {
     color_t button_cl = is_selected ? back_color : COLOR_GRAY;
     color_t text_cl = is_selected ? COLOR_BLACK : COLOR_WHITE;
+    if (GuiDefaults::RadioButtonCornerRadius) {
+        display::DrawRect(rc_btn, parent_color);
+        rc_btn += Rect16::Left_t(GuiDefaults::RadioButtonCornerRadius);
+        rc_btn -= Rect16::Width_t(2 * GuiDefaults::RadioButtonCornerRadius);
+    }
     render_text_align(rc_btn, text, pf, button_cl, text_cl, { 0, 0, 0, 0 }, Align_t::Center());
 }
 
 bool RadioButton::IsEnabled() const {
-    return responses ? (*responses)[0] != Response::_none : false; //faster than cnt_responses(responses)!=0
+    return responseFromIndex(0) != Response::_none; //faster than cnt_responses(responses)!=0
```

Comment:
Does this comment still apply? Moreover, if `cnt_responses` was constexpr (which IMO could be), the whole `IsEnabled` may boil down to something like 3 instructions.

Reply:
comment is still valid
`cnt_responses `use `std::find` and it is not `constexpr` in C++17
also responses are assigned at runtime, so it would not work anyway

Reply:
I have a constexpr find_if somewhere

Reply:
ok